gather

```diff
@@ -29,22 +24,15 @@ strip_wildcard = re.compile("^(\*.)+")
 strip_redacted = re.compile("^(\?\.)+")
 
 
-def run(options=None):
+def run(options=None, cache_dir="./cache", results_dir="./results"):
```

Comment:
It looks like `"./cache"` appears over 20 times in this PR, generally specifying a default value, which seems unwieldy.

It's also not the right level to be specifying values all over the place. The `cache_dir` and `results_dir` must always be subdirectories of the `--output` directory (which defaults to `.`). There's no way to specify separate `cache/` and `results/` locations, and they can always be derived as a direct subdirectory of `--output`.

So given that, what I would expect to see is the default `--output` value specified once, and then cache/results directories dynamically calculated as needed from that value. If there's concern about using global variables, then this value could be stored in a more intentional place and imported (a top-level module where config values/constants are drawn from? some other kind of config file for project values/defaults?).

Reply:
It is messy to have it in a lot of places, but I wanted to use the explicit & repetitive approach first to replace the use of global variables that end up being accessed from relatively deep in the execution path, and make sure that our tests cover their being passed along correctly, before moving to cleaner options.

Reply:
A cleaner approach might look like something I was working on for gatherers and that I just uploaded as https://github.com/18F/domain-scan/pull/229